include/nbl/asset/utils/CHLSLCompiler.h

```diff
@@ -0,0 +1,59 @@
+// Copyright (C) 2018-2022 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_ASSET_C_HLSL_COMPILER_H_INCLUDED_
+#define _NBL_ASSET_C_HLSL_COMPILER_H_INCLUDED_
+
+#include "nbl/asset/utils/ISPIRVOptimizer.h"
+#include "nbl/asset/utils/IShaderCompiler.h"
+
+namespace nbl::asset
+{
+
+class NBL_API2 CHLSLCompiler final : public IShaderCompiler
+{
+	public:
+		IShader::E_CONTENT_TYPE getCodeContentType() const override { return IShader::E_CONTENT_TYPE::ECT_HLSL; };
+
+		CHLSLCompiler(core::smart_refctd_ptr<system::ISystem>&& system);
+
+		struct SOptions : IShaderCompiler::SOptions
+		{
+			// TODO: Add extra dxc options
+
+			void setCommonData(const IShaderCompiler::SOptions& opt)
+			{
+				static_cast<IShaderCompiler::SOptions&>(*this) = opt;
+			}
```

Comment:
you can hoist this to `IShaderCompiler::SOptions` btw, a derived `SOptions` will downcast itself just fine

Reply:
yea good point 😄